src/lsm/segmented_array.zig

```diff
@@ -879,6 +973,58 @@ fn TestContext(
             }
 
             try context.remove_all();
+
+            if (element_order == .unsorted) {
+                // Insert every element at the beginning.
+                while (context.array.len() < element_count_max) {
+                    try context.insert_before_first();
+                }
+                assert(context.array.node_count >= TestArray.node_count_max - 1);
+                {
+                    var node: u32 = 0;
+                    while (node < context.array.node_count) : (node += 1) {
+                        if (node == 0) {
+                            // The first node is more than half-full.
+                            assert(context.array.count(node) > @divExact(TestArray.node_capacity, 2));
+                        } else {
+                            // All other nodes are half-full.
+                            assert(context.array.count(node) == @divExact(TestArray.node_capacity, 2));
+                        }
+                    }
+                }
+
+                // Remove all-but-one elements from the last node and insert them into the first node.
+                const element_count_last = context.array.count(context.array.node_count - 1);
+                var element_index: usize = 0;
+                while (element_index < element_count_last - 1) : (element_index += 1) {
+                    try context.remove_last();
+                    try context.insert_before_first();
+                }
+
+                // We can only reach the worst-case layout in some configurations.
```

Comment:
@sentientwaffle @jorangreef This is the actual bound on node_count_max. Is it ok to just test it like this, or would you prefer to shrink node_count_max in the cases where this layout can't be reached?

Reply:
Eugh, I didn't place the comment very well. Here's the full test - https://github.com/tigerbeetledb/tigerbeetle/blob/1bf2e99a8465f49195428ec5b7a72dafc6ee8a51/src/lsm/segmented_array.zig#L978-L1026

Reply:
Since you calculate the exact upper bound anyway, I would probably just use that.

Reply:
Ok, done.